thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -253,7 +255,23 @@ public Builder registerAction(
                 @NonNull String alias,
                 @NonNull String actionName,
                 @NonNull Class<? extends Action> actionClass){
-            //TODO: validate actionClass, should contain only 1 field
+            // validate field of actionClass
+            List<Field> fields = new ArrayList<>();
+            for (Field field : actionClass.getDeclaredFields()) {
+                if (!field.getName().equals("this$0")){ // in case of inner class, has this field
```

Comment:
@shinsuke-nara 
> Test parent = new Test();
> AInnerClasss child =  parent.new AInnerClasss();

In the code, why it require you to create Test instance first, because method `main` is static. If you don't use static method. You can directly create AInnerClass.